### PR TITLE
Beautify carousel

### PR DESCRIPTION
**Pull request by BenGWeeks**
A small amendment to the TabView in the ImageCarousel to implement a custom "CarouselDotsView" that places dots below the set of images.

https://user-images.githubusercontent.com/5950171/233879699-6e728de5-a4f0-44bb-bca4-16ed42706f15.mp4

Possible enhancements:

- Add 1/3 to the top left of the images as per Figma proposal from Sasha and Roberto.
- Update how to deal with different sizes of images.

### CODE REVIEW

**Comment by jb55** [comment]
yay!

File: damus/Components/ImageCarousel.swift [hunk 1]
```diff
@@ -73,41 +73,73 @@ struct ImageCarousel: View {
...
+            .fullScreenCover(isPresented: $open_sheet) {
+                ImageView(urls: urls, disable_animation: disable_animation)
+            }
+            .frame(height: height)
+            .onTapGesture {
+                open_sheet = true
+            }
+            .onChange(of: selectedIndex) { value in
+                            selectedIndex = value
+                        }
```
Comment (jb55) [inline damus/Components/ImageCarousel.swift:115]: this looks like a no-op to me

**Review by jb55 — CHANGES_REQUESTED** [review]

**Comment by BenGWeeks** [comment]
Do you want 1/3 in the top left like the designs by the way?

**Comment by jb55** [comment]
ah nah I'm not a fan of the 1/3. sorry I have a huge backlog of things to review and it is very time consuming. getting to this now

**Comment by BenGWeeks** [comment]
No worries. Was gonna update this to a setting. Been looking at lots of improvements to the carousel in general too.

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by dmnyc** [comment]
Just a note that if the first image in the carousel contains a white background, the dot UI is invisible.
note10z2ud6f6rutmr8wm50qdzafrvkqrj9am64x8nhd7x8ln7nekqnes0r63m2

**Comment by BenGWeeks** [comment]
This has now been updated. Carousel counter moved to a setting and default is off.

File: damus/Components/ImageCarousel.swift (hunk 1 above)
Comment (BenGWeeks) [reply damus/Components/ImageCarousel.swift]: Not sure why, but it doesn't work if I remove this.

**Comment by BenGWeeks** [comment]
Does need testing on a physical device as I saw something quirky in the emulator with a mouse., but not sure if a mouse + emulator issue or something else.

**Comment by jb55** [comment]
On Fri, May 26, 2023 at 02:25:34PM -0700, Ben Weeks wrote:
>Does need testing on a physical device as I saw something quirky in the emulator with a mouse., but not sure if a mouse + emulator issue or something else.

testing now

**Comment by jb55** [comment]
Merged in d0915434 . I made a bunch of changes.

- Removed settings and indicator. I don't think we need this.
- Restored fit/fill aspect ratios for each image
- Restored blurhash

**Pull request closed without merging**